Loop/Managers/WatchDataManager.swift

```diff
@@ -168,13 +168,13 @@ final class WatchDataManager: NSObject {
     private func createWatchContext(recommendingBolusFor potentialCarbEntry: NewCarbEntry? = nil, _ completion: @escaping (_ context: WatchContext) -> Void) {
         let loopManager = deviceManager.loopManager!
 
-        let glucose = loopManager.glucoseStore.latestGlucose
-        let reservoir = loopManager.doseStore.lastReservoirValue
+        let glucose = deviceManager.glucoseStore.latestGlucose
+        let reservoir =  deviceManager.doseStore.lastReservoirValue
         let basalDeliveryState = deviceManager.pumpManager?.status.basalDeliveryState
 
-        loopManager.getLoopState { (manager, state) in
+        loopManager.getLoopState { [weak deviceManager] (manager, state) in
```

Comment:
I think switching to a `weak deviceManager` is going to cause problems in this function. For example, there are a number of `updateGroup.enter()` invocations that will *not* be matched with a corresponding `updateGroup.leave()` if `deviceManager` is `nil` (since the functions with completion handlers that eventually call `updateGroup.leave()` will never be called).

Plus, if `deviceManager` is `nil`, then the end result will be a mostly incomplete (and likely useless) context.

I think you need a guard at the top to check if deviceManager is `nil` (and, perhaps, get a strong reference to do away with all of the `?` below). However, you'll have to consider what you do in this guard. The completion handler needs to be called, but returning an empty `WatchContext` may not be the best choice. Might need to update the completion handler to accept an *optional* `WatchContext` and let downstream worry about a missing context.

@ps2 Thoughts?

Reply:
Good catch, Darin. @novalegra what was the reason to switch to weak deviceManager?

Reply:
I was trying to avoid a retain loop, but that might not be a concern here?

Reply:
Yeah, I don't see a retain loop here.

Reply:
I'll revert back to a strong reference